extension/src/background/store.ts

```diff
@@ -2,9 +2,60 @@ import { combineReducers } from "redux";
 import { configureStore } from "@reduxjs/toolkit";
 
 import { sessionSlice } from "background/ducks/session";
+import {
+  SESSION_STORAGE_ENABLED,
+  dataStorageAccess,
+  sessionStorage,
+  browserStorage,
+} from "./helpers/dataStorage";
 
-export const store = configureStore({
+// Session storage can be used to persist redux store in Manifest v3 because service workers go idle after 30 seconds
+// Session storage is not currently supported in Firefox, which blocks our migration to using this by default
+const REDUX_STORE_KEY = "redux-store";
+const sessionStore = dataStorageAccess(
+  SESSION_STORAGE_ENABLED ? sessionStorage : browserStorage,
```

Comment:
correct me if I'm wrong - but it looks like `sessionStore` will never be saved to `browserStorage` as we're just using redux when `SESSION_STORAGE_ENABLED` is false (`loadState` and `saveStore` are only used when `SESSION_STORAGE_ENABLED` === true)

If that's the case, I wonder if we shouldn't even bother having `browserStorage` as a backup. Especially since we considered `browserStorage` to be a risky place to save private creds.

Reply:
Hmm yeah I was trying to not risk using the session store when it's not supported but I think it's probably ok to do what you said since it won't actually be used in the case where session storage is disabled.